File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TJSONNumber.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

public final class TJSONNumber extends TJSONValue {
	protected Double value;

	/**
	 * Class constructor, initialized the internal value to null
	 */
	public TJSONNumber() {
		super();
		value = null;
	}

	/**
	 * Gets the internal value.
	 * 
	 * @return
	 */
	public Double getValue() {
		if (value == null)
			return null;
		else
			return value;
	}

	/**
	 * Class constructor, initialized the internal value by parse the String
	 * passed
	 * 
	 * @param value
	 */
	public TJSONNumber(String value) {
		super();
		this.value = Double.valueOf(value);
	}

	/**
	 * Class constructor, initialized the internal value with the double passed
	 * 
	 * @param value
	 */
	public TJSONNumber(double value) {
		super();
		this.value = new Double(value);
	}

	/**
	 * Class constructor, initialized the internal value with the long passed
	 * 
	 * @param value
	 */
	public TJSONNumber(long value) {
		super();
		this.value = new Double(value);
	}

	/**
	 * Class constructor, initialized the internal value with the int passed
	 * 
	 * @param value
	 */
	public TJSONNumber(int value) {
		super();
		this.value = new Double(value);
	}

	public Object getInternalObject() {
		return value;
	}

	public String toString() {
		if (value != null)
			return value.toString();
		return NullString;
	}

	public int getJsonValueType() {
		return JSONValueType.JSONNumber;
	}
}

File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TJSONNumber.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TJSONNumber.h"


@implementation TJSONNumber

-(id) init{
	self = [super init];
	if (self) {
	//	value = [[NSNumber alloc ]init ];
		value = nil;
	}
	return self;
}
-(void) dealloc {
	[value release];
	[super dealloc];
}
-(id) initWithNumber:(NSNumber*)aValue{
	self =[self init];
	if (self) {
		[value release];
		value = [aValue retain];
	}
	return self;
	

}
-(id) initWithString:(NSString *) aValue {
	self =[self init];
	if (self) {
		[value release];		
		NSNumberFormatter * f = [[NSNumberFormatter alloc] init];
		[f setNumberStyle:NSNumberFormatterDecimalStyle];
		value = [[f numberFromString:aValue]retain];
		[f release];
		
	}
	return self;
}

-(id) initWithDouble:(double) aValue {
	self =[self init];
	if (self) {
		[value release];
		value = [[NSNumber numberWithDouble:aValue]retain];
	}
	return self;
}

-(id) initWithLong:(long) aValue {
	self =[self init];
	if (self) {
		[value release];
		value = [[NSNumber numberWithLong:aValue] retain];
	}
	return self;
}
-(id) initWithInt:(int) aValue {
	self =[self init];
	if (self) {
		[value release];
		value = [[NSNumber numberWithInt:aValue] retain];
	}
	return self;
}

-(id) getInternalObject{
	return value;
}

-(NSString *) toString{
	if (value) {
		return [value stringValue];
	}

    return [self nullString];	
}
-(JSONValueType) getJSONValueType{
	return JSONNumber;
}
-(NSNumber *) getValue{
	return value;
}

@end
@implementation TJSONNumber(jsonnumberCreation)
+ (id) jsonNumberWithNumber:(NSNumber *) value{
	return [[[TJSONNumber alloc]initWithNumber:value]autorelease];
}
@end
